functionary/builder/management/commands/generate_package_schema.py

```diff
@@ -0,0 +1,42 @@
+import json
+import pathlib
+
+from django.core.management.base import BaseCommand
+from drf_jsonschema_serializer import to_jsonschema
+
+from builder.api.v1.serializers.package_definition import (
+    PackageDefinitionWithVersionSerializer,
+)
+
+
+class Command(BaseCommand):
+    help = "Generates the schema.json used to validate the package.yaml"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "-o",
+            "--output",
+            action="store",
+            dest="schema_file",
+            type=pathlib.Path,
+            nargs=1,
+            help="Output file to write the schema to",
+        )
+
+    def handle(self, *args, **options):
+        json_schema = to_jsonschema(PackageDefinitionWithVersionSerializer())
+        json_schema["$schema"] = "https://json-schema.org/draft/2020-12/schema"
```

Comment:
Not related to this line, but throwing it here...

It would be nice if we could tack on `"additionalProperties": false` at both the `package` and `functions.items` levels.  That would make it so that if someone adds a property that is not defined in the schema, they get yelled at for it.  Right now they'll only see warnings about required properties that are missing.

Reply:
Added to the `schema.json`. The `to_jsonschema` doesn't take parameters and the alternative was a deep reach many levels into the dict.